src/controllers/hid/hidcontroller.cpp

```diff
@@ -243,13 +244,32 @@ int HidController::close() {
 bool HidController::poll() {
     Trace hidRead("HidController poll");
 
-    int result = hid_read(m_pHidDevice, m_pPollData, sizeof(m_pPollData) / sizeof(m_pPollData[0]));
-    if (result == -1) {
-        return false;
-    } else if (result > 0) {
-        Trace process("HidController process packet");
-        QByteArray outData(reinterpret_cast<char*>(m_pPollData), result);
-        receive(outData, mixxx::Time::elapsed());
+    int result = 1;
+    while (result > 0) {
+        // Rotate between two buffers so the memcmp below does not require deep copying to another buffer.
+        unsigned char* pCurrentBuffer = m_pPollData[m_iPollingBufferIndex];
+        if (m_iPollingBufferIndex == 0) {
+            m_iPollingBufferIndex = 1;
+        } else {
+            m_iPollingBufferIndex = 0;
+        }
```

Comment:
We could shorten this to:
```suggestion
     m_iPollingBufferIndex = (m_iPollingBufferIndex + 1) % 2;
```
But it's also okay as-is.